WooCommerce/Classes/POS/Presentation/Order Messages/PointOfSaleOrderSyncCouponsErrorMessageView.swift

```diff
@@ -0,0 +1,89 @@
+           let nsAttributedString = try? NSAttributedString(
+               data: data,
+               options: [.documentType: NSAttributedString.DocumentType.html],
+               documentAttributes: nil) {
+            var attributedString = AttributedString(nsAttributedString)
+            attributedString.font = POSFontStyle.posBodyLargeRegular().font()
+            attributedString.foregroundColor = UIColor(Color.posOnSurface)
+            return attributedString
+        }
+        return AttributedString(message)
+    }
+
+    var body: some View {
+        HStack(alignment: .center) {
+            Spacer()
+            VStack(alignment: .center, spacing: POSSpacing.none) {
+                Spacer()
+                POSErrorExclamationMark(size: .large)
+                Spacer().frame(height: PointOfSaleCardPresentPaymentLayout.imageAndTextSpacing)
+                VStack(alignment: .center, spacing: PointOfSaleCardPresentPaymentLayout.textSpacing) {
+                    Text("Invalid coupons")
+                        .foregroundStyle(Color.posOnSurface)
+                        .font(.posHeadingBold)
+
+                    Text(attributedMessage)
+                        .padding([.leading, .trailing])
+                }
+                Spacer().frame(height: PointOfSaleCardPresentPaymentLayout.textAndButtonSpacing)
+                Button("Continue without coupons", action: {
+                    posModel.removeAllCouponsFromCart()
+                    retryHandler()
+                })
+                .buttonStyle(POSFilledButtonStyle(size: .normal))
+                .padding([.leading, .trailing], Constants.buttonSidePadding)
+                .padding([.bottom], Constants.buttonBottomPadding)
+
+                Spacer()
+            }
+            .multilineTextAlignment(.center)
+            Spacer()
+        }
+    }
+}
+
+@available(iOS 17.0, *)
+private extension PointOfSaleOrderSyncCouponsErrorMessageView {
+    enum Constants {
+        static let headerSpacing: CGFloat = POSSpacing.large
+        static let textSpacing: CGFloat = POSSpacing.medium
+        static let buttonSidePadding: CGFloat = POSPadding.xxLarge
+        static let buttonBottomPadding: CGFloat = POSPadding.medium
+    }
+}
+
+// MARK: - TODO when copy is finalized
+//
+//private extension PointOfSaleOrderSyncErrorMessageView {
+//    enum Localization {
+//        static let title = NSLocalizedString(
+//            "pointOfSale.orderSync.couponsError.title",
+//            value: "Invalid coupons",
+//            comment: "Title of the error when failing to validate coupons and calculate order totals"
+//        )
+//
+//        static let actionTitle = NSLocalizedString(
+//            "pointOfSale.orderSync.couponsError.proceed",
+//            value: "Continue without coupons",
+//            comment: "Button title to remove coupons and retry synchronizing order and calculating order totals"
+//        )
+//    }
+//}
```

Comment:
Let's add a follow-up somewhere and a link here so we don't forget :D

Reply:
👍 Okay, I will. All these PRs are with dummy UI so we'll need to do a second round to create a proper look,

Reply:
- Created a task https://github.com/woocommerce/woocommerce-ios/issues/15424
- Added a comment https://github.com/woocommerce/woocommerce-ios/pull/15420/commits/6376ff75c620e28833dc74d581db6b2cd804acf7